### PR TITLE
Contact form

### PR DESCRIPTION
**Pull request by nirgn975**
#257

### CODE REVIEW

**Comment by nirgn975** [comment]
We're running tests now, so.. (;

File: client/src/app/components/contact/contact.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  checkAll(){
+      if(!this.contactForm.valid) {
+          this.displayErrs = true;
+      }
+  }
+  buildForm() {
+      this.contactForm = this.formBuilder.group({
+          name: this.formBuilder.control(null, [Validators.minLength(2),
+                                                Validators.required]),
```
Comment (nirgn975) [inline client/src/app/components/contact/contact.component.ts]: Just FYI: We're working with GitHub style, so each line can be 120 characters (On Django too)

**Comment by DavidBronfen** [comment]
- What's the default color for the enabled button? I guess it won't be this yellow...
- What about the error messages, is there any resource to get the content of the messages?
(and other missed content of the titles and so)
open_pension_

-Another thing, the `checkAll()` method is triggered on `mouseenter` event, why?
Don't we need to trigger it on 'submit' or on 'focusout'

**Comment by DavidBronfen** [comment]
@nirgn975 
- Why Travis Node.js 6 build is failing while Node.js 8 is passing.
- Another thing, I did some changes in the contact style but now we have the point of the logic, validation and error messages and I found a nice way to do it.
[Here](https://coryrylan.com/blog/angular-form-builder-and-validation-management) is the post that shows it, take a look and let me know if you think that it's a good way for us so I will change our form to use this method.

File: client/src/app/components/contact/contact.component.scss [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  flex-direction: column;
+  width: 40%;
+  margin-right: 10%;
+}
+
+form {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  // flex-basis: auto;
```
Comment (nirgn975) [inline client/src/app/components/contact/contact.component.scss]: Remove this comment

File: client/src/app/components/contact/contact.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+    this.nameControl = this.contactForm.get('name');
+    this.emailControl = this.contactForm.get('email');
+    this.contentControl = this.contactForm.get('content');
+  }
+
+  checkError(event) {
+    if (event.keyCode === 9) {
+      return;
```
Comment (nirgn975) [inline client/src/app/components/contact/contact.component.ts]: Add a comment for the next person to know 9 is `tab` key and we pass the check when move between inputs with the `tab` key.

File: client/src/app/components/contact/contact.component.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      case 'content': {
+        this.ContentError = true;
+        break;
+      }
+    }
+  }
+
+  onSubmit() {
+    const formModel = this.contactForm.value;
+    console.log(formModel);
```
Comment (nirgn975) [inline client/src/app/components/contact/contact.component.ts]: Cleanup the `console.log()`.

**Pull request merged**